### PR TITLE
Docs: Renamed References to Reference in site-src

### LINKED ISSUE #2469: Docs: Rename references to reference in site-src
**What happened**:
Clicking on the "Reference" tab leads to a 404 (https://gateway-api.sigs.k8s.io/reference/implementers-guide.md).

**What you expected to happen**:
To go to https://gateway-api.sigs.k8s.io/references/implementers-guide/.

**Anything else we need to know?**:
Until https://github.com/kubernetes-sigs/gateway-api/pull/2470, we had broken navigation due to inconsistency between the name used in our docs and the directory used in our docs. We should rename `site-src/references` to `site-src/reference`, and update `mkdocs.yml` to link to the new pages and add redirects for all the old links.

### PR DESCRIPTION
**Pull request by murthy95**
**What type of PR is this?**
/kind documentation

-->
**What this PR does / why we need it**:
This PR renames `site-src/references` to `site-src/reference`, and add redirects for all the old links.

**Which issue(s) this PR fixes**:
Fixes  #2469

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -77,15 +77,15 @@ nav:
     - gRPC Routing: guides/grpc-routing.md
     - Migrating from Ingress: guides/migrating-from-ingress.md
   - Reference:
-    - Implementer's Guide: references/implementers-guide.md
+    - Implementer's Guide: reference/implementers-guide.md
```
Comment (robscott) [inline mkdocs.yml:80]: Can you add some corresponding entries to `redirect_maps` above?

**Review by robscott — COMMENTED** [review]
Thanks for the work on this @murthy95!

/ok-to-test

**Comment by robscott** [comment]
May not be obvious, but clicking "details" beside the `deploy/netlify` presubmit will take you to a deploy preview where you can test this out:

https://deploy-preview-2471--kubernetes-sigs-gateway-api.netlify.app/

File: mkdocs.yml (hunk 1 above)
Comment (murthy95) [reply mkdocs.yml:83]: Done

**Comment by robscott** [comment]
Thanks @murthy95!

/ok-to-test
/lgtm
/approve

**Pull request merged**